server/src/main/java/io/crate/breaker/CellsSizeEstimator.java

```diff
@@ -57,6 +60,24 @@ protected long estimateSize(int valueCount, IntFunction<Object> values) {
         };
     }
 
+    public static CellsSizeEstimator estimateAsStrings() {
+        return new CellsSizeEstimator() {
+            @Override
+            protected long estimateSize(int valueCount, IntFunction<Object> values) {
+                long size = 0;
+                ObjectMapper mapper = new ObjectMapper();
+                for (int i = 0; i < valueCount; i++) {
+                    try {
+                        size += mapper.writeValueAsString(values.apply(i)).length();
+                    } catch (JsonProcessingException e) {
+                        throw new RuntimeException(e);
```

Comment:
I think this approach be slow, some alternative approches could be:
- Wrap `XContent` builder/generator such that they can account for the memory usage(lazily)?
- `DataType.valueBytes()` returns number of bytes that the value takes up; extend this to accept a custom counting logic?